gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/Resolver.kt

```diff
@@ -89,11 +89,12 @@ class Resolver(
     revision: String,
     currentCoordinates: Map<Coordinate.Key, Coordinate>,
   ): Configuration {
-    // Kotlin deps anywhere in the hierarchy are a special case we'll handle later
-    val kotlinDeps = { dependency: ExternalDependency -> dependency.group == "org.jetbrains.kotlin" && dependency.version != null }
+    // Kotlin deps anywhere in the hierarchy are a special case we'll handle later separately, unless this configuration
+    // has no parents (such as the classpath conf, which contains plugins, including kotlin ones), in which case handle those deps here
+    val kotlinDeps = { dependency: ExternalDependency -> (dependency.group?.startsWith("org.jetbrains.kotlin") ?: false) && dependency.version != null }
     val latest = configuration.allDependencies
       .filterIsInstance<ExternalDependency>()
-      .filterNot(kotlinDeps)
+      .filter { configuration.hierarchy.size > 1 && kotlinDeps(it) }
```

Comment:
maybe `filterNot`? trying locally

Reply:
passes except for your new test, which I think might be a formatting difference?

```console
PluginUpdateDetectionSpec > kotlin plugin in the implementation configuration is properly handled FAILED
    Condition not satisfied:

    result.output.contains """The following dependencies have later milestone versions: - org.jetbrains.kotlin:kotlin-gradle-plugin [$KOTLIN_VERSION -> """
    |      |      |                                                                                                                   |
    |      |      false                                                                                                               1.6.0
    |
    |      > Task :dependencyUpdates
    |
    |      ------------------------------------------------------------
    |      : Project Dependency Updates (report to plain text file)
    |      ------------------------------------------------------------
    |
    |      The following dependencies are using the latest milestone version:
    |       - org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.0
    |       - org.jetbrains.kotlin:kotlin-stdlib:1.6.0
```

Reply:
I think detecting by `isForced` might be what's needed.